src/IIIFPresentation/Services/Manifests/ManifestItemsParser.cs

```diff
@@ -174,22 +177,38 @@ private CanvasPainting CreatePartialCanvasPainting(ResourceBase? resource,
         return cp;
     }
     
-    private string? TryGetValidCanvasId(int customerId, Canvas currentCanvas)
+    private string? TryGetValidCanvasId(int customerId, Canvas currentCanvas, 
+        List<CanvasPainting>? paintedResourceCanvasPaintings)
     {
         if (currentCanvas.Id == null) return null;
 
         if (!Uri.TryCreate(currentCanvas.Id, UriKind.Absolute, out var canvasId))
         {
-            CanvasHelper.CheckForProhibitedCharacters(currentCanvas.Id);
+            currentCanvas.Id = CanvasHelper.CheckForProhibitedCharacters(currentCanvas.Id, logger, false);
+            
+            if (currentCanvas.Id != null && !(paintedResourceCanvasPaintings?.Any(cp => cp.Id == currentCanvas.Id) ?? false))
```

Comment:
`paintedResourceCanvasPaintings?.Any(cp => cp.Id == xx) ?? false` is repeated twice - make helper method?

Would allow for 
```cs
// if (currentCanvas.Id != null && !(paintedResourceCanvasPaintings?.Any(cp => cp.Id == currentCanvas.Id) ?? false))
if (currentCanvas.Id != null && !HasMatchingThing(currentCanvas.Id))

// if (!(paintedResourceCanvasPaintings?.Any(cp => cp.Id == checkedCanvasId) ?? false))
if (!HasMatchingThing(checkedCanvasId))
``